web/controllers/user_controller.ex

```diff
@@ -34,6 +35,23 @@ defmodule CodeCorps.UserController do
     end
   end
 
+  def github_connect(conn, code) do
+    current_user = Guardian.Plug.current_resource(conn)
+    with {:ok, user} <- Github.connect(current_user, code)
```

Comment:
>Now after saying that, Im wondering if it is better (conventional really) to update the Repo in the Github.connect function or in this github_connect function? ¯\ _ (ツ) _ /¯

We already have `Github.associate`, which does the associating part, so either `Github.connect` can call `Github.associate` internally, or we can call both in order. I'm fine with either way.

>Also @begedin, think we should send the state as well to github_connect?

The `state` part is dealt with in the first step. I don't think the api should have to deal with it. Basically:

Client
- redirects user to github with `state`
- user accepts connection
- github redirects back to client with `code` and the same `state`
- client checks the original and received state are the same, if they are, posts code to `/github-connect`

That being said, using that approach means we'd have to store the original state clientside somehow, maybe cookie or local storage, in order to check they match. Alternatively, we'd have to do something like

Client
- asks API to generate `state` and assigns to current user
- redirects user to github with `state`
- user accepts connection
- github redirects back to client with `code` and the same `state`
- client posts `code` and `state` to `/github-connect`
- API checks state, continues with the regular process

In this case, we'd need an extra endpoint to generate and assign the `state` server-side. It's probably safer, but also more complex.